### PR TITLE
fix(toxav): fix multiple logic bugs in audio module

### PR DESCRIPTION
**Pull request by iphydf**
- Fix heap-buffer overflow in `ac_iterate` (PLC) by validating sampling rates and channel counts from untrusted RTP packets.
- Fix out-of-bounds read by adding explicit packet length validation (minimum 5 bytes).
- Fix jitter buffer reset bug where late packets caused a full buffer clear due to unsigned wrap-around.
- Fix jitter buffer PLC logic to correctly handle sequence number wrap-around at 16-bit boundaries.
- Refactor `ac_iterate` to process all currently ready packets in a single call, improving throughput.
- Optimize `ac_iterate` by moving the large scratch buffer allocation outside the packet processing loop.
- Add unit tests covering jitter buffer edge cases, wrap-around scenarios, and malicious packet handling.
- Refactor `ACSession` to be opaque and provide accessor for frame duration.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2935)

### CODE REVIEW

File: toxav/audio.h [hunk 1]
```diff
@@ -34,40 +35,26 @@
...
-ACSession *ac_new(Mono_Time *mono_time, const Logger *log, ToxAV *av, uint32_t friend_number,
-                  toxav_audio_receive_frame_cb *cb, void *cb_data);
+typedef void ac_audio_receive_frame_cb(uint32_t friend_number, const int16_t *pcm, size_t sample_count,
+                                       uint8_t channels, uint32_t sampling_rate, void *user_data);
+
+typedef struct ACSession ACSession;
+
+struct RTPMessage;
+
+ACSession *ac_new(Mono_Time *mono_time, const Logger *log, uint32_t friend_number,
```
Comment (Green-Sky) [inline toxav/audio.h:45]: while we are at it, we can use the friend number type here too

Comment (iphydf) [reply toxav/audio.h:45]: No we can't. This file does not depend on tox.h.
